src/transformState.js

```
@@ -4,8 +4,31 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
-function transformState(state, actions) {
-  // write code here
-}
+  function transformState(state, actions) {
+    for (const action of actions) {
+      switch (action.type) {
+        case 'addProperties':
+          Object.assign(state, action.extraData);
+          break;
+  
+        case 'removeProperties':
+          for (const key of action.keysToRemove) {
+            delete state[key];
+          }
+          break;
+  
+        case 'clear':
+          for (const key in state) {
+            delete state[key];
+          }
+          break;
+  
+        default:
+          return 0;
```

let's throw the error in default